src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWareHouse.java

```diff
@@ -0,0 +1,196 @@
+
+    /**
+     * The list of deliverymen registered to this building
+     */
+    private static final List<BlockPos> registeredDeliverymen = new ArrayList<>();
+
+    /**
+     * The tileEntity of the building.
+     */
+    private TileEntityWareHouse tileEntity;
+
+    /**
+     * Instantiates a new warehouse building.
+     *
+     * @param c the colony.
+     * @param l the location
+     */
+    public BuildingWareHouse(final Colony c, final BlockPos l)
+    {
+        super(c, l);
+    }
+
+    /**
+     * Register deliveryman with the warehouse.
+     * @param buildingWorker the building of the worker.
+     * @return true if able to register or already registered
+     */
+    public boolean registerWithWareHouse(BuildingDeliveryman buildingWorker)
+    {
+        if(registeredDeliverymen.contains(buildingWorker.getID()))
+        {
+            return true;
+        }
+
+        if(registeredDeliverymen.size() >= getBuildingLevel())
+        {
+            return false;
+        }
+
+        registeredDeliverymen.add(buildingWorker.getID());
+        return true;
+    }
+
+    /**
+     * Check if deliveryman is allowed to access warehouse.
+     * @param buildingWorker
+     * @return
+     */
+    public boolean canAccessWareHouse(BuildingDeliveryman buildingWorker)
+    {
+        if(registeredDeliverymen.contains(buildingWorker.getID()))
+        {
+            return true;
+        }
+        return false;
+    }
+
+    @NotNull
+    @Override
+    public String getSchematicName()
+    {
+        return WAREHOUSE;
+    }
+
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return 3;
+    }
+
+    @Override
+    public void readFromNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.readFromNBT(compound);
+        final NBTTagList deliverymanTagList = compound.getTagList(TAG_DELIVERYMAN, Constants.NBT.TAG_COMPOUND);
+        for (int i = 0; i < deliverymanTagList.tagCount(); i++)
+        {
+            final BlockPos pos = NBTUtil.getPosFromTag(deliverymanTagList.getCompoundTagAt(i));
+            if(getColony() != null && getColony().getBuilding(pos) instanceof AbstractBuildingWorker)
+            {
+                registeredDeliverymen.add(pos);
+            }
+        }
+    }
+
+    @Override
+    public void writeToNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.writeToNBT(compound);
+        @NotNull final NBTTagList levelTagList = new NBTTagList();
+        for (@NotNull final BlockPos deliverymanBuilding : registeredDeliverymen)
+        {
+            levelTagList.appendTag(NBTUtil.createPosTag(deliverymanBuilding));
+        }
+        compound.setTag(TAG_DELIVERYMAN, levelTagList);
+    }
+
+    /**
+     * Returns the tile entity that belongs to the colony building.
+     *
+     * @return {@link TileEntityColonyBuilding} object of the building.
+     */
+    @Override
+    public TileEntityWareHouse getTileEntity()
+    {
+        Colony colony = getColony();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'colony' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)